tests/testsuite/pub_priv.rs

```diff
@@ -246,3 +246,184 @@ fn workspace_dep_made_public() {
         .masquerade_as_nightly_cargo(&["public-dependency"])
         .run()
 }
+
+#[cargo_test(nightly, reason = "exported_private_dependencies lint is unstable")]
+fn allow_priv_in_tests() {
```

Comment:
Looks like proc-macros are lib's but I'm assuming those shouldn't be checked with `exported_private_dependencies`.

Example libs are an odd case.

I'm more mixed about crate types, like cdylibs.

@ehuss any thoughts?

Reply:
> Another one we should cover is build scripts.

OK, add a test for custom build script

Reply:
I think it would be good if all libs warn just like a normal rlib. Any lib type can expose types within item signatures, and it seems like it would be good hygiene to be explicit about that. 

For proc-macros, it is not possible to have public items (other than the proc-macros themselves), so it is not possible to "leak" anything AFAIK. But I wouldn't go out of my way to try to disable it there.

Reply:
I'll go for that.  We can always re-visit.